hnn_core/network.py

```diff
@@ -1253,23 +1253,62 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
 
         self.connectivity.append(deepcopy(conn))
 
-    def clear_connectivity(self):
-        """Remove all connections defined in Network.connectivity
+    def clear_connectivity(self, src_types="all"):
+        """Remove connections with src_type in Network.connectivity.
+
+        Parameters
+        ----------
+        src_types : list | all | drives | local
+            Source types of connections to be cleared
+            all - Clear all connections (Default)
+            drives - Clear connections originating from external drives
+            local - Clear connections within cells
+
         """
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] in self.external_drives.keys():
-                connectivity.append(conn)
-        self.connectivity = connectivity
-
-    def clear_drives(self):
-        """Remove all drives defined in Network.connectivity"""
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] not in self.external_drives.keys():
-                connectivity.append(conn)
-        self.external_drives = dict()
-        self.connectivity = connectivity
+
+        if src_types == "all":
+            src_types = list()
+            for conn in self.connectivity:
+                if conn['src_type'] not in src_types:
+                    src_types.append(conn['src_type'])
+        elif src_types == "drives":
+            src_types = self.drive_names
+        elif src_types == "local":
+            src_types = list()
+            external_drives = self.drive_names
+            for conn in self.connectivity:
+                if (conn['src_type'] not in src_types and
+                   conn['src_type'] not in external_drives):
+                    src_types.append(conn['src_type'])
```

Comment:
same here ... I think you just need to do a list comprehension to leave out the external drives from all the cell populations